runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -522,11 +517,46 @@ public static StreamingDataflowWorker fromOptions(DataflowWorkerHarnessOptions o
         workFailureProcessor,
         streamingCounters,
         memoryMonitor,
-        configFetcherComputationStateCacheAndWindmillClient.windmillStreamFactory(),
+        dependencies.windmillStreamFactory(),
         Executors.newSingleThreadScheduledExecutor(
             new ThreadFactoryBuilder().setNameFormat("RefreshWork").build()),
         stageInfo,
-        configFetcherComputationStateCacheAndWindmillClient.windmillDispatcherClient());
+        dependencies.windmillDispatcherClient(),
+        dependencies.channelCache(),
+        dependencies.stubFactory());
+  }
+
+  private static Dependencies.Builder initialDependencies(
+      DataflowWorkerHarnessOptions options, ComputationConfig.Fetcher configFetcher) {
+    if (options.getUseWindmillIsolatedChannels() == null
+        || options.getIsWindmillServiceDirectPathEnabled()) {
+      ConfigAwareChannelFactory channelFactory =
+          new ConfigAwareChannelFactory(options.getWindmillServiceRpcChannelAliveTimeoutSec());
+      ChannelCache channelCache = ChannelCache.create(channelFactory);
+      ChannelCachingRemoteStubFactory stubFactory =
+          ChannelCachingRemoteStubFactory.create(options.getGcpCredential(), channelCache);
+      GrpcDispatcherClient dispatcherClient = GrpcDispatcherClient.create(stubFactory);
+      configFetcher
+          .getGlobalConfigHandle()
+          .registerConfigObserver(
+              config -> {
+                if (channelFactory.tryConsumeJobConfig(config)) {
+                  dispatcherClient.reloadDispatcherEndpoints(config.windmillServiceEndpoints());
+                }
+              });
+      return Dependencies.builder()
+          .setChannelCache(channelCache)
+          .setStubFactory(stubFactory)
+          .setWindmillDispatcherClient(dispatcherClient);
+    } else {
+      ChannelCache channelCache = createConfigAwareChannelCache(options, configFetcher);
```

Comment:
can we just always set things up to be config aware and push the different on if it is isolated/directpath enabled down to where we react to changes?

I think that will simplify things in this file where we have a lot of different ways we wire things up already